polaris-core/src/main/java/org/apache/polaris/core/persistence/transactional/PolarisMetaStoreManagerImpl.java

```diff
@@ -1980,11 +1983,22 @@ private PolarisEntityResolver resolveSecurableToRoleGrant(
               String.valueOf(
                   Integer.parseInt(properties.getOrDefault(PolarisTaskConstants.ATTEMPT_COUNT, "0"))
                       + 1));
-          task.setEntityVersion(task.getEntityVersion() + 1);
-          task.setProperties(PolarisObjectMapperUtil.serializeProperties(callCtx, properties));
-          ms.writeEntity(callCtx, task, false, originalTask);
+          updatedTask.setProperties(
+              PolarisObjectMapperUtil.serializeProperties(callCtx, properties));
+          EntityResult result = updateEntityPropertiesIfNotChanged(callCtx, ms, null, updatedTask);
```

Comment:
Question: why do we need to update task entities in a "load" call?

Reply:
Yeah, `loadTasks` probably is just a bad name. It is more correctly `leaseTasks`. Should I rename in this PR?

Reply:
I think a rename would be good for the sake of clarity, but we can do it in a separate PR :+1: thx!